packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -1126,156 +1150,257 @@ export class InteractionServer implements ProtocolHandler<MatterDevice>, Interac
-                                endpoint,
-                                receivedWithinTimedInteraction,
-                            ),
-                        StatusResponseError,
-                        async error => {
-                            const errorLogText = `Error ${Diagnostic.hex(error.code)}${
-                                error.clusterCode !== undefined ? `/${Diagnostic.hex(error.clusterCode)}` : ""
-                            } while invoking command: ${error.message}`;
-                            if (error instanceof ValidationError) {
-                                logger.info(
-                                    `Validation-${errorLogText}${error.fieldName !== undefined ? ` in field ${error.fieldName}` : ""}`,
-                                );
-                            } else {
-                                logger.info(errorLogText);
-                            }
-                            return {
-                                code: error.code,
-                                clusterCode: error.clusterCode,
-                                responseId: command.responseId,
-                                response: TlvNoResponse.encodeTlv(),
-                            };
-                        },
-                    );
-                    const { code, clusterCode, responseId, response } = result;
-                    if (response.length === 0) {
-                        invokeResponses.push({
-                            status: { commandPath: path, status: { status: code, clusterStatus: clusterCode } },
-                        });
-                    } else {
-                        invokeResponses.push({
-                            command: {
-                                commandPath: { ...path, commandId: responseId },
-                                commandFields: response,
-                            },
+                    continue;
+                }
+                if (
+                    getMatterModelClusterCommand(clusterId, commandId)?.fabricScoped &&
+                    (!exchange.session.isSecure || !(exchange.session as SecureSession<MatterDevice>).fabric)
+                ) {
+                    logger.debug(`This invoke requires a secure session with a fabric assigned which is missing.`);
+                    if (isConcretePath) {
+                        await processResponseResult({
+                            status: { commandPath: path, status: { status: StatusCode.UnsupportedAccess }, commandRef },
                         });
                     }
+                    continue;
                 }
-            }),
-        );
 
-        // TODO support suppressResponse for responses
-        return {
-            suppressResponse: false,
-            interactionModelRevision: INTERACTION_MODEL_REVISION,
-            invokeResponses,
-        };
+                const result = await tryCatchAsync(
+                    async () =>
```

Comment:
Maybe we could overload `tryCatchAsync` take a promise or a function?  Then `async () => await this.invokeCommand(...)` could become `this.invokeCommand(...)` and it'd remove an anonymous stack trace frame.  Even if not this could just be `() => this.invokeCommand(...)`.

But actually, if we just didn't use tryCatchAsync here I don't think it'd be a big hit to readability and it'd remove two stack frames and (I think) at least two promises from every invoke.

I've been going out of my way to avoid stack frames where reasonable in critical code paths...  Mainly so stack truncation removes less useful information but it has perf effect too (admittedly prolly marginal in network code).